tests/mock_update_catalog.py

```diff
@@ -0,0 +1,120 @@
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+Mock data for catalog_update unit tests
+"""
+
+from kubernetes.client.rest import ApiException
+import datetime
+import os
+from unittest import mock
+from tests.mocks import COS_VERSIONS
+
+# Mocking environment variables before import so that:
+# 1. import (create_config_map, update_config_map and main) is successful
+# 2. Additionally they will be used in testcase to verify the tests.
+mock.patch.dict(
+    os.environ, {
+        'PRODUCT': 'sat',
+        'PRODUCT_VERSION': '1.0.0',
+        'YAML_CONTENT_STRING': 'Test data',
+        'CONFIG_MAP_NAMESPACE': 'myNamespace'
+    }
+).start()
+
+UPDATE_DATA = {
+    '2.0.0': {
+        'component_versions': {
+            'docker': [
+                {'name': 'cray/cray-cos', 'version': '1.0.0'},
+                {'name': 'cray/cos-cfs-install', 'version': '1.4.0'}
+            ]
+        }
+    }
+}
+
+ERR_NOT_FOUND = 404
+
+
+class Name:
+    """
+    Class to provide dummy metadata object with name and resource_version
+    """
+    def __init__(self):
+        self.name = 'cray-product-catalog'
+        self.resource_version = 1
+
+
+class Response:
+    """
+    Class to generate response for k8s api call api_instance.read_namespaced_config_map(name, namespace)
+    """
+    def __init__(self):
+        self.data = COS_VERSIONS
+        self.metadata = Name()
+
+
+class ApiException(ApiException):
+    """
+    Custom Exception to define status
+    """
+    def __init__(self):
+        self.status = ERR_NOT_FOUND
+
+
+class ApiInstance():
+    """
+    Class to raise custom exception and ignore function calls
+    """
+    def __init__(self, raise_exception=False):
+        self.raise_exception = raise_exception
+        self.count = 0
+
+    def create_namespaced_config_map(self, namespace='a', body='b'):
+        """
+        Dummy Function to raise exception, if needed
+        """
+        if self.raise_exception:
+            raise ApiException()
+        else:
+            pass
+
+    def read_namespaced_config_map(self, name, namespace):
+        """
+        Dummy Function to :
+        1. Raise exception
+        2. generate and return proper response with data and metadata
+        """
+        # if this is called for first time return exception, so that product cm is created.
+        if self.count == 0:
+            self.count += 1
+            raise ApiException()
+        else:
+            return Response()

```

Comment:
No need for this to be in an `else` clause, since the `if` part always ends with an exception being raised.
```suggestion
        return Response()
```